azure/scope/machinepool.go

```diff
@@ -58,12 +57,14 @@ type (
 
 	// MachinePoolScope defines a scope defined around a machine pool and its cluster.
 	MachinePoolScope struct {
-		logr.Logger
-		client           client.Client
-		patchHelper      *patch.Helper
-		MachinePool      *capiv1exp.MachinePool
-		AzureMachinePool *infrav1exp.AzureMachinePool
 		azure.ClusterScoper
+		logr.Logger
+		AzureMachinePool   *infrav1exp.AzureMachinePool
+		MachinePool        *capiv1exp.MachinePool
+		client             client.Client
+		patchHelper        *patch.Helper
+		vmssState          *infrav1exp.VMSS
```

Comment:
`ScaleSet.Reconcile` sets the state of the VMSS on the scope so that the scope can properly update the status of the `AzureMachinePool`. I don't think the information would be useful to set directly on the `AzureMachinePool`, but it does lead to more concise information being set on the `AzureMachinePool`, like conditions, etc.